File: src/qiuyi/Array/BestTimetoBuyandSellStock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.empty()) return 0;
        int maxVal = 0;
        int minVal = prices[0];
        
        for (int i = 1; i < prices.size(); i++) {
            minVal = min(minVal, prices[i]);
            maxVal = max(maxVal, prices[i]-minVal);
        }
        return maxVal;
    }
};

File: src/haoxiang/Array/BestTimetoBuyandSellStock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        minNum = None
        maxNum = None
        res = 0
        for i in prices:
            if minNum is None or i < minNum:
                minNum = maxNum = i
            elif i>maxNum:
                maxNum = i

            res = max(res,maxNum-minNum)
        return res